File: morning_exercises/day4.py
def twoSum(nums: list, target: int)->list:
    res = []
    seen = {}

    if (len(nums)==0): return []

    for i in range(0, len(nums)):
        complement = target - nums[i]

        if(complement in seen): res.append([seen[complement], i])

        seen[nums[i]] = i

    return res

print(twoSum([1,2,3,4],5))

File: morning_exercises/day4.cs
public static class TwoSumOptimal
{
    public static List<List<int>> twoSum(List<int> nums, int target)
    {
        List<List<int>> res = [];
        Dictionary <int, int> seen = [];

        if (nums.Count==0) return [];

        for (int i=0; i<nums.Count; i++)
        {
            int complement = target - nums[i];

            if (seen.TryGetValue(complement, out int value)) res.Add(new List<int> {value, i});

            seen[nums[i]] = i;
        }
        return res;
    }
}